### ISSUE TITLE
value of loss function (mse) change whether I compile the model inside or outside a function

### ISSUE THREAD

**Issue by Guigo95** [keras-team/keras#14010]
Hello, 

 I get different loss (with mse) (at first epoch 1300 vs 30) when I compile my model inside a definition function (1) or juste before the fit in the main.
(1)
def_model():
    ....
    model = Model(input = inputs, output = outputs )   
    model.compile
return model

model = def_model()
model.fit

(2)
def_model():
     ....
    model = Model(input = inputs, output = outputs )   
return model

model = def_model()
model.compile
model.fit

Someone has an explanation?

Thanks

**Comment by Saduf2019** [comment]
@Guigo95 
Moving this issue to closed status as there has been no activity, in case you still face the error please create a new issue.
